istioctl/cmd/analyze.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"istio.io/istio/galley/pkg/config/analysis/analyzers"
+	"istio.io/istio/galley/pkg/config/analysis/local"
+	"istio.io/istio/galley/pkg/config/processor/metadata"
+	"istio.io/istio/galley/pkg/source/kube/client"
+	"istio.io/pkg/log"
+
+	"github.com/spf13/cobra"
+)
+
+// Analyze command
+func Analyze() *cobra.Command {
+	analysisCmd := &cobra.Command{
+		Use:   "analyze <file|globpattern>...",
+		Short: "Analyze Istio configuration and print validation messages",
+		Example: `
+# Analyze yaml files
+istioctl experimental analyze a.yaml b.yaml
+
+# Analyze the current live cluster
+istioctl experimental analyze -c $HOME/.kube/config
+
+# Analyze the current live cluster, simulating the effect of applying additional yaml files
+istioctl experimental analyze -c $HOME/.kube/config a.yaml b.yaml
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			loggingOptions.SetOutputLevel("processing", log.ErrorLevel)
+			loggingOptions.SetOutputLevel("source", log.ErrorLevel)
```

Comment:
What's the reason for changing the default log levels for these scopes?

Reply:
The default log levels for the processing components are pretty chatty, and the result is a lot of irrelevant output in the terminal. The verbosity makes sense for galley logs, but not so much when we're running the same components on the command line. 

I'm not thrilled about overwriting log levels for a particular command, any ideas on how we could do this better?

Reply:
Changing the log levels seems reasonable. Can you add a short comment to explain this to future readers?

Reply:
Sure, done.